### PR TITLE
Breakout editor

### PR DESCRIPTION
**Pull request by karolina-rachuta**
PD bonus task:
client/default/js/experience/editors/com/sfcc/customPinEditorBreakoutScript
client/default/js/experience/editors/com/sfcc/customPinEditorTriggerScript
client/default/js/helpers/hotspotsImageHelper
client/default/js/pinBreakoutScript
client/default/js/pinTriggerScript
client/default/js/hotspotScript
experience/components/commerce_assets/heroBanner.json
experience/components/commerce_assets/heroBanner.js
templates/default/components/commerce_assets/heroBanner.isml

### CODE REVIEW

File: cartridges/app_custom_breakout/cartridge/client/default/js/experience/editors/com/sfcc/customCarEditorBreakoutScript.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Code in the client-side JavaScript file for the breakout  (BeeIt-DEMO)
+(function () {
+    subscribe('sfcc:ready', function () {
+            // Once the breakout editor is ready, the custom code is able to select or
+            // Create a value. Any meaningful change to a value/selection needs to be
+            // reflected back into the host container via a `sfcc:value` event.
+            var template = document.createElement('template');
+            template.innerHTML = "<div><p>Yugo</p><p>Mercedes</p><p>BMW</p><p>Seat</p></div>";
```
Comment (kalipiev) [inline cartridges/app_custom_breakout/cartridge/client/default/js/experience/editors/com/sfcc/customCarEditorBreakoutScript.js:8]: This is demo text ?

File: cartridges/app_custom_breakout/cartridge/client/default/js/experience/editors/com/sfcc/customCarEditorBreakoutScript.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            var clone = document.importNode(template.content, true);
+            document.body.appendChild(clone);
+            var openButtonEl = document.querySelector('p');
+            // openButtonEl.addEventListener('click', handleSelect);
```
Comment (kalipiev) [inline cartridges/app_custom_breakout/cartridge/client/default/js/experience/editors/com/sfcc/customCarEditorBreakoutScript.js:12]: Please remove commented code

File: cartridges/app_custom_breakout/cartridge/client/default/js/experience/editors/com/sfcc/customCarEditorBreakoutScript.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                paragraph.addEventListener('click', handleSelect);
+            });
+        });
+
+    function handleSelect({ target }) {
+
+        // The value changed and the breakout editor's host is informed about the
+        // value update via a `sfcc:value` event.
+        const selectedValue = target.innerText;
+        console.log(selectedValue);
```
Comment (kalipiev) [inline cartridges/app_custom_breakout/cartridge/client/default/js/experience/editors/com/sfcc/customCarEditorBreakoutScript.js:24]: console.log

File: cartridges/app_custom_breakout/cartridge/client/default/js/experience/editors/com/sfcc/customCarEditorTriggerScript.js
Comment (kalipiev) [inline cartridges/app_custom_breakout/cartridge/client/default/js/experience/editors/com/sfcc/customCarEditorTriggerScript.js:1]: Please try avoiding hard-coded text in the javascript. It's better to be taken from a template instead.

File: cartridges/app_custom_breakout/cartridge/experience/editors/com/sfcc/customCarEditorTrigger.js
Comment (kalipiev) [inline cartridges/app_custom_breakout/cartridge/experience/editors/com/sfcc/customCarEditorTrigger.js:1]: This file is redundant I believe?

File: cartridges/plugin_interactive_images/cartridge/client/default/js/experience/editors/com/sfcc/imageHotspotEditorTriggerScript.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
+
+// Code in the client-side JavaScript file for the trigger editor
+(() => {
+    let container = document.createElement('div');
+    let hotspotsDisplay = document.createElement('span');
+    let currentValue = {};
+
+    function renderHotspots(hotspots) {
+        console.log("renderHotspots-trigger", renderHotspots);
+        console.log("renderHotspots-trigger-hotspots", hotspots);
```
Comment (kalipiev) [inline cartridges/plugin_interactive_images/cartridge/client/default/js/experience/editors/com/sfcc/imageHotspotEditorTriggerScript.js:10]: console.log

File: cartridges/plugin_interactive_images/cartridge/client/default/js/experience/editors/com/sfcc/imageHotspotEditorTriggerScript.js [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+        hotspots.forEach((hotspot, index) => {
+            hotspotHtml += `<span class="hotspot-breakout-marker" data-index="${index + 1} - " style="display: block; margin-top: 5px; margin-left: 15px">${hotspot[2]}</span>`;
+        });
+
+        return hotspotHtml;
+    }
+
+    function showCreatedHotspots(currentValue) {
+        console.log("showCreatedHotspots-trigger", showCreatedHotspots); 
+        console.log("showCreatedHotspots-trigger-currentValue", currentValue);
```
Comment (kalipiev) [inline cartridges/plugin_interactive_images/cartridge/client/default/js/experience/editors/com/sfcc/imageHotspotEditorTriggerScript.js:21]: same here

File: cartridges/plugin_interactive_images/cartridge/client/default/js/experience/editors/com/sfcc/imageHotspotEditorTriggerScript.js [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+        document.body.appendChild(container);
+
+
+        showCreatedHotspots(currentValue);
+    });
+
+    subscribe('sfcc:value', async ({ breakout, ...rest }) => {
+        console.log("subscribe-second-trigger", subscribe); 
+        console.log("subscribe-second-trigger-breakoutValue", breakout); 
+        console.log("subscribe-second-trigger-rest", ...rest);
```
Comment (kalipiev) [inline cartridges/plugin_interactive_images/cartridge/client/default/js/experience/editors/com/sfcc/imageHotspotEditorTriggerScript.js:76]: same here

File: cartridges/plugin_interactive_images/cartridge/client/default/js/helpers/hotspotsImageHelper.js
Comment (kalipiev) [inline cartridges/plugin_interactive_images/cartridge/client/default/js/helpers/hotspotsImageHelper.js:1]: All the files within plugin_interactive_images I think are redundant

File: cartridges/app_custom_breakout/cartridge/client/default/js/experience/editors/com/sfcc/customCarEditorBreakoutScript.js (hunk 1 above)
Comment (karolina-rachuta) [reply cartridges/app_custom_breakout/cartridge/client/default/js/experience/editors/com/sfcc/customCarEditorBreakoutScript.js:8]: yes, I will delete

File: cartridges/app_custom_breakout/cartridge/client/default/js/experience/editors/com/sfcc/customPinEditorBreakoutScript.js [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    let listTitile;
+    let id = 0;
+    var currentValue = {};
+    var newPinSettings;
+
+    
+    subscribe('sfcc:ready', async (value) => {
+        currentValue = Object.assign({}, value.value);
+        if (value) {
+        imageFromTrigger = document.createElement('img');
```
Comment (kalipiev) [inline cartridges/app_custom_breakout/cartridge/client/default/js/experience/editors/com/sfcc/customPinEditorBreakoutScript.js:16]: indentation

File: cartridges/app_custom_breakout/cartridge/client/default/js/experience/editors/com/sfcc/customPinEditorBreakoutScript.js [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+            newPinSettings = {
+                xCoordinate: xCoordinate,
+                yCoordinate: yCoordinate,
+                id: pin.dataset.id
+            }
+
+            //updating currentValue
+            if (!currentValue.pins) {
+            currentValue.pins = [];
```
Comment (kalipiev) [inline cartridges/app_custom_breakout/cartridge/client/default/js/experience/editors/com/sfcc/customPinEditorBreakoutScript.js:66]: indentation

File: cartridges/app_custom_breakout/cartridge/client/default/js/experience/editors/com/sfcc/customPinEditorBreakoutScript.js [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+        } else {
+            console.error('Image element with class "image_PD" not found.');
+        }
+    }
+
+    function handlePinRemoval(event) {
+        var allElementsToRemove = document.querySelectorAll('.pin-remove');
+        var deleteButtonId = event.target.dataset.id;
+        allElementsToRemove.forEach((element)=> {
+            if(element.dataset.id === deleteButtonId) {
```
Comment (kalipiev) [inline cartridges/app_custom_breakout/cartridge/client/default/js/experience/editors/com/sfcc/customPinEditorBreakoutScript.js:98]: indentation

File: cartridges/app_custom_breakout/cartridge/experience/components/commerce_assets/heroBanner.json [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+                    "type": "file",
+                    "required": true
+                },
+                {
+                    "id": "alt",
+                    "name": "Banner Image Alt Text",
+                    "description": "The image alt text shown by the banner.",
+                    "type": "string",
+                    "required": false
+                },
```
Comment (kalipiev) [inline cartridges/app_custom_breakout/cartridge/experience/components/commerce_assets/heroBanner.json:24]: Please add a new boolean attribute which to indicate if the image to be loaded lazy or not.

**Pull request merged**